cni/network/network.go

```diff
@@ -941,77 +941,89 @@ func (plugin *NetPlugin) Delete(args *cniSkel.CmdArgs) error {
+				log.Printf("[cni-net] Failed to query network:%s: %v", networkID, err)
+				// Log the error but return success if the network is not found.
+				// if cni hits this, mostly state file would be missing and it can be reboot scenario where
+				// container runtime tries to delete and create pods which existed before reboot.
+				err = nil
+				return err
+			}
+		}
 
-	// Query the network.
-	if nwInfo, err = plugin.nm.GetNetworkInfo(networkID); err != nil {
-		if !nwCfg.MultiTenancy {
-			log.Printf("[cni-net] Failed to query network:%s: %v", networkID, err)
-			// Log the error but return success if the network is not found.
-			// if cni hits this, mostly state file would be missing and it can be reboot scenario where
-			// container runtime tries to delete and create pods which existed before reboot.
+		endpointID := GetEndpointID(args)
+		// Query the endpoint.
+		if epInfo, err = plugin.nm.GetEndpointInfo(networkID, endpointID); err != nil {
+			if !nwCfg.MultiTenancy {
+				// attempt to release address associated with this Endpoint id
+				// This is to ensure clean up is done even in failure cases
+				log.Printf("[cni-net] Failed to query endpoint %s: %v", endpointID, err)
+				logAndSendEvent(plugin, fmt.Sprintf("Release ip by ContainerID (endpoint not found):%v", args.ContainerID))
+				if err = plugin.ipamInvoker.Delete(nil, nwCfg, args, nwInfo.Options); err != nil {
+					return plugin.RetriableError(fmt.Errorf("failed to release address(no endpoint): %w", err))
+				}
+			}
+			// Log the error but return success if the endpoint being deleted is not found.
 			err = nil
 			return err
 		}
-	}
 
-	endpointID := GetEndpointID(args)
-	// Query the endpoint.
-	if epInfo, err = plugin.nm.GetEndpointInfo(networkID, endpointID); err != nil {
+		// schedule send metric before attempting delete
+		defer sendMetricFunc() //nolint:gocritic
+		logAndSendEvent(plugin, fmt.Sprintf("Deleting endpoint:%v", endpointID))
+		// Delete the endpoint.
+		if err = plugin.nm.DeleteEndpoint(networkID, endpointID); err != nil {
+			// return a retriable error so the container runtime will retry this DEL later
+			// the implementation of this function returns nil if the endpoint doens't exist, so
+			// we don't have to check that here
+			return plugin.RetriableError(fmt.Errorf("failed to delete endpoint: %w", err))
+		}
 
 		if !nwCfg.MultiTenancy {
-			// attempt to release address associated with this Endpoint id
-			// This is to ensure clean up is done even in failure cases
-			log.Printf("[cni-net] Failed to query endpoint %s: %v", endpointID, err)
-			logAndSendEvent(plugin, fmt.Sprintf("Release ip by ContainerID (endpoint not found):%v", args.ContainerID))
-			if err = plugin.ipamInvoker.Delete(nil, nwCfg, args, nwInfo.Options); err != nil {
-				return plugin.RetriableError(fmt.Errorf("failed to release address(no endpoint): %w", err))
+			// Call into IPAM plugin to release the endpoint's addresses.
+			for i := range epInfo.IPAddresses {
```

Comment:
how about this?
`for _, ipAddr := range epInfo.IPAddresses {`

Reply:
Hi Tamilmani, I like this style. I think it's old code there and if we change this to for _, ipAddr type, it will have the linter issue => "Implicit memory aliasing in for loop", so I suggest we can keep this as it is.......